streaming-connect-sink/src/main/java/com/adobe/platform/streaming/sink/impl/AEPPublisher.java

```diff
@@ -100,7 +103,14 @@ public void publishData(List<Pair<String, SinkRecord>> messages) throws AEPStrea
         for (JsonNode messageResponse : publishMessagesResponses) {
           if (messageResponse.hasNonNull(STATUS_KEY)) {
             failedMessageCount++;
-            LOG.debug("Failed to publish message to Adobe Experience Platform: {}", messageResponse);
+            final Pair<String, SinkRecord> failedMessage = messages.get(getFailedMessageIndex(messageResponse));
+            LOG.debug("Failed to publish message: {} to Adobe Experience Platform due to the error: {}",
+              failedMessage, messageResponse);
+            if (Objects.nonNull(errorReporter)) {
+              final int responseCode = messageResponse.get(STATUS_KEY).asInt();
+              errorReporter.report(failedMessage.getRight(),
+                new HttpException(String.format("error response= %s", failedMessage), responseCode));
```

Comment:
failedMessage is already provided to errorReported. In place of failedMessage inlet failure message should be added.
```suggestion
                new HttpException(String.format("error response= %s", <message>), responseCode));
```